cloud/store/recover.go

```diff
@@ -0,0 +1,183 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/raft"
+	raftbolt "github.com/hashicorp/raft-boltdb/v2"
+	"github.com/vmihailenco/msgpack"
+)
+
+// Peer represents information about a peer in the Raft cluster
+type Peer struct {
+	Address  string `msgpack:"Address" json:"address"`
+	ID       string `msgpack:"ID" json:"id"`
+	Suffrage int64  `msgpack:"Suffrage" json:"-"`
+	NonVoter bool   `msgpack:"-" json:"non_voter"`
+}
+
+// genPeersFileFromBolt checks if there was an existing raft.db and has persisted config
```

Comment:
Thanks for the great godoc, it's something we need to improve in Weaviate's code base and it makes the review easier.